### PR TITLE
Replaced exception with the warnings and removed related failing specs(used earlier for raising issue)

### PR DESCRIPTION
**Pull request by sanga1794**
Signed-off-by: sanga17 <sausekar@msystechnologies.com>

## Description
When uploading vault item that contains \n errors with the below message.

Check that backslashes are escaped with another backslash (e.g. C:\Windows) in double-quoted strings.
Looking at that error, we changed the \n to \n and the error stops BUT the content vault item actually have \n and no new line is inserted.

Before Fix: 
```
bundle exec knife vault update test test1 -J ../../chef-repo/root.json -c ../../chef-repo/.chef/knife.rb
ERROR: ChefVault::Exceptions::InvalidValue: Value 'my
     passwords' of key 'text1' contains non-printable characters. Check that backslashes are escaped with another backslash (e.g. C:\\Windows) in double-quoted strings.

bundle exec knife vault show test test1 -c ../../chef-repo/.chef/knife.rb
id:       test1
text1:    my\npasswords
username: root

```
After Fix:
``` 
bundle exec knife vault update test test1 -J ../../chef-repo/root.json -c ../../chef-repo/.chef/knife.rb
WARN: Value 'my
passwordsj' of key 'text1' contains non-printable characters. Check that backslashes are escaped with another backslash (e.g. C:\\Windows) in double-quoted strings.


bundle exec knife vault show test test1 -c ../../chef-repo/.chef/knife.rb
id:       test1
text1:    my
passwords
username: root
```
## Related Issue
Fixed: https://github.com/chef/chef-vault/issues/366

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/chef/knife/mixin/helper.rb [hunk 1]
```diff
@@ -59,7 +59,7 @@ def validate_json(json)
             next unless printable?(value.to_s)
```
Comment (lamont-granquist) [inline lib/chef/knife/mixin/helper.rb:59]: shouldn't printable just be changed to also match whitespace?

Comment (sanga1794) [reply lib/chef/knife/mixin/helper.rb:59]: Hi @lamont-granquist , for the whitespace it is working fine without any issue as shwon below
```
json file content:
{
	"id": "test1",
	"text1": "my passwords",
	"username": "root"
}

bundle exec knife vault update test test1 -J ../../chef-repo/root.json -c ../../chef-repo/.chef/knife.rb

bundle exec knife vault show test test1 -c ../../chef-repo/.chef/knife.rb
id:       test1
text1:    my passwords
username: root


```
Based on this is there a need to add warnings for whitespace? It should be be fine here to not add warning IMO. Please suggest.

Comment (lamont-granquist) [reply lib/chef/knife/mixin/helper.rb:59]: I'm talking about adding `[[:space:]]` to the match for `printable?` which also matches `\n` and `\r` and `\t` in addition to ascii space and various unicode spaces.

Comment (lamont-granquist) [reply lib/chef/knife/mixin/helper.rb:59]: you probably want to just use `String#valid_encoding?` though.  or use the `String#encode!` method to convert the string to UTF-8 but raise on any errors rather than replacing.

Comment (sanga1794) [reply lib/chef/knife/mixin/helper.rb:59]: > space

okk, will add `[[:space]]`. Thanks @lamont-granquist !

Comment (sanga1794) [reply lib/chef/knife/mixin/helper.rb:59]: > I'm talking about adding `[[:space:]]` to the match for `printable?` which also matches `\n` and `\r` and `\t` in addition to ascii space and various unicode spaces.

Okk, will add `[[:space:]]` Thanks @lamont-granquist !

**Comment by sanga1794** [comment]
Hi @lamont-granquist, I have added the regex for whitesapce, Please have look into it. Thanks!!

**Comment by TerrieB1** [comment]
Can we get an update on this case for the customer. Thank you

**Comment by sanga1794** [comment]
@lamont-granquist Please have look into the latest changes, Thanks!!

**Comment by moritoki-chef** [comment]
@sanga1794 Hi. I often use chef-vault and am having trouble with this problem. This problem(pull requiest) seems to have stopped for a long time, is there any problem? I hope for a quick solution. :-)

File: lib/chef/knife/mixin/helper.rb [hunk 2]
```diff
@@ -59,7 +59,7 @@ def validate_json(json)
...
 
             msg = "Value '#{value}' of key '#{key}' contains non-printable characters. Check that backslashes are escaped with another backslash (e.g. C:\\\\Windows) in double-quoted strings."
-            raise ChefVault::Exceptions::InvalidValue, msg
+            ChefVault::Log.warn(msg)
```
Comment (vkarve-chef) [inline lib/chef/knife/mixin/helper.rb:62]: i believe an Exception still needs to be raised as before, except that the allowed characters have expanded.

File: spec/chef/helper_spec.rb [hunk 3]
```diff
@@ -5,18 +5,13 @@
   include ChefVault::Mixin::Helper
 
   let(:json_data) { '{"username": "root", "password": "abcabc"}' }
-  let(:json_data_control_char) { '{"username": "root", "password": "abc\abc"}' }
   let(:buggy_json_data) { '{"username": "root", "password": "abc\abc"' }
```
Comment (vkarve-chef) [inline spec/chef/helper_spec.rb:8]: will be good to add tests that ensure we allow for tabspace, whitespace and newline going forward.

**Review by vkarve-chef — APPROVED** [review]

**Review by vkarve-chef — APPROVED** [review]

**Review by vkarve-chef — APPROVED** [review]
